apollo-ios/Sources/Apollo/AnyGraphQLResponse.swift

```diff
@@ -0,0 +1,100 @@
+#if !COCOAPODS
+import ApolloAPI
+#endif
+
+/// An abstract GraphQL response used for full and incremental responses.
+struct AnyGraphQLResponse {
+  public let body: JSONObject
+
+  private let rootKey: CacheReference
+  private let variables: GraphQLOperation.Variables?
+
+  init(
+    body: JSONObject,
+    rootKey: CacheReference,
+    variables: GraphQLOperation.Variables?
+  ) {
+    self.body = body
+    self.rootKey = rootKey
+    self.variables = variables
+  }
+
+
+  func execute<
+    Accumulator: GraphQLResultAccumulator,
+    Operation: GraphQLOperation
+  >(
+    selectionSet: any SelectionSet.Type,
+    in operation: Operation.Type,
+    with accumulator: Accumulator
+  ) throws -> Accumulator.FinalResult? {
+    guard let dataEntry = body["data"] as? JSONObject else {
+      return nil
+    }
+
+    return try executor.execute(
+      selectionSet: selectionSet,
+      in: Operation.self,
+      on: dataEntry,
+      withRootCacheReference: rootKey,
+      variables: variables,
+      accumulator: accumulator
+    )
+  }
+
+  func execute<
+    Accumulator: GraphQLResultAccumulator,
+    Data: RootSelectionSet
+  >(
```

Comment:
We should try to narrow the scope of when this can be called so it's not inadvertently called on the root of an operation. Maybe make the `Data` have a `Deferrable` requirement?

If not possible, we should at least document the use cases and warn that this should not be used when executing an entire operation.

Reply:
Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/4ee07f0a97e89fe044d90a5570acc5f7a9c6c18e and https://github.com/apollographql/apollo-ios-dev/pull/217/commits/98e61d10ba84783ccc37de1ddf1d68b7cebc55ff.